src/metrology_apis/__init__.py

```diff
@@ -8,8 +8,36 @@
 __all__ = ["__version__", "Dimension", "Quantity", "Unit"]
 
 
+@runtime_checkable
+class MetrologyNamespace[Q: Quantity[V, U, D], V, U: Unit[D], D: Dimension](Protocol):
+
+    @staticmethod
+    def asdimension(obj: str | D) -> D: ...
+
+    @staticmethod
+    def asunit(obj: str | U) -> U: ...
+
+    @staticmethod
+    def asquantity(obj: Q | V, *, unit: U) -> Q: ...
+
+
 @runtime_checkable
 class Dimension(Protocol):
+    def __metrology_namespace__[Q: Quantity, U: Unit](
+        self, /, *, api_version: str | None = None
+    ) -> MetrologyNamespace[Q, U, Self]:
+        """
+        Returns an object that has all the metrology API functions on it.
+        Parameters
+        ----------
+        api_version: str or None
+            string representing the version of the metrology API specification to be returned. If it is ``None``, it should return the namespace corresponding to latest version of the metrology API specification.  If the given version is invalid or not implemented for the given module, an error should be raised. Default: ``None``.
```

Comment:
```suggestion
            string representing the version of the metrology API specification to be returned. If it is `None`, it should return the namespace corresponding to latest version of the metrology API specification.  If the given version is invalid or not implemented for the given module, an error should be raised. Default: `None`.
```

Sphinx and tools can link to `None`, but display ``None`` as code.